### PR TITLE
chore: improves filter API by 

### PR DESCRIPTION
**Pull request by jcchavezs**
This PR improves the filter API, most likely the one around traceable filter by:
0. Includes start/stop API to traceable filter.
1. Delegate the internal state retrieval into the filter appender: this allows to have a more cohesive internal state API (as opposed to the internal config API) mainly because it is very likely that those retrieving the internal config to do the filter injection are going to also declare functions to be executed at the end of the app lifecycle (i.e. we plug things automatically on start but want to plug things at the end too).
2. Appender also look into blocking config to decide whether to inject the blocking filter or not.

### CODE REVIEW

File: filter/traceable/filter.go [hunk 1]
```diff
@@ -28,33 +28,62 @@ const defaultAgentManagerEndpoint = "localhost:5441"
...
+	started        bool
 }
 
-var _ filter.Filter = (*libTraceableFilter)(nil)
+var _ filter.Filter = (*Filter)(nil)
+
+// Start() starts the threads to poll config
+func (f *Filter) Start() bool {
+	if f.blockingEngine == nil {
+		return false
```
Comment (jcchavezs) [inline filter/traceable/filter.go:59]:
```suggestion
```

File: _tests/Dockerfile.test [hunk 2]
```diff
@@ -6,15 +6,21 @@ FROM traceable_goagent_test_base:${TRACEABLE_GOAGENT_DISTRO_VERSION}
 
 WORKDIR ${GOPATH}/src/github.com/Traceableai/goagent
 
-COPY . .
+COPY go.mod go.mod
```
Comment (jcchavezs) [inline _tests/Dockerfile.test:9]: This allows to cache dependency pull steps.

File: filter/traceable/filter.go [hunk 3]
```diff
@@ -28,33 +28,63 @@ const defaultAgentManagerEndpoint = "localhost:5441"
 const defaultPollPeriodSec = 30
 
 // NewFilter creates libtraceable based blocking filter
-func NewFilter(config *traceableconfig.AgentConfig) filter.Filter {
+func NewFilter(config *traceableconfig.AgentConfig) *Filter {
 	blockingConfig := config.BlockingConfig
 	// disabled if no blocking config or enabled is set to false
-	if blockingConfig == nil || blockingConfig.Enabled.Value == false {
-		return filter.NoopFilter{}
+	if blockingConfig == nil ||
```
Comment (davexroth) [inline filter/traceable/filter.go:34]: Log this

File: filter/traceable/filter.go [hunk 4]
```diff
@@ -28,33 +28,63 @@ const defaultAgentManagerEndpoint = "localhost:5441"
...
 
 	libTraceableConfig := getLibTraceableConfig(config)
 	defer freeLibTraceableConfig(libTraceableConfig)
 
-	var blockingFilter libTraceableFilter
+	var blockingFilter Filter
 	ret := C.traceable_new_blocking_engine(libTraceableConfig, &blockingFilter.blockingEngine)
 	if ret != C.TRACEABLE_SUCCESS {
-		return filter.NoopFilter{}
+		return &Filter{}
```
Comment (davexroth) [inline filter/traceable/filter.go:46]: Log this

File: filter/traceable/filter.go (hunk 3 above)
Comment (jcchavezs) [reply filter/traceable/filter.go:34]: We don't having logging support in place so only option might be std logging. In the other hand we don't expect people to call this function by themselves (we plug this automatically) so I think it is better if we return an error.

File: filter/traceable/filter.go (hunk 4 above)
Comment (jcchavezs) [reply filter/traceable/filter.go:46]: Same as above https://github.com/Traceableai/goagent/pull/11/files#r715073538

**Review by davexroth — APPROVED** [review]

**Pull request merged**